app/models/v3/persistence/isolation_segment_model.rb

```diff
@@ -1,20 +1,32 @@
 module VCAP::CloudController
   class IsolationSegmentModel < Sequel::Model(:isolation_segments)
+    SHARED_ISOLATION_SEGMENT_GUID = '933b4c58-120b-499a-b85d-4b6fc9e2903b'.freeze
+
     include Serializer
     ISOLATION_SEGMENT_MODEL_REGEX = /\A[[:print:]]+\Z/
 
     one_to_many :spaces,
       key: :isolation_segment_guid,
       primary_key: :guid
 
+    many_to_many :organizations,
+      left_key: :isolation_segment_guid,
+      left_primary_key: :guid,
+      right_key: :organization_guid,
+      right_primary_key: :guid,
+      join_table: :organizations_isolation_segments,
+      after_add: proc { |isolation_segment, org| org.default_isolation_segment(isolation_segment) },
+      before_remove: proc { |isolation_segment, org| org.unset_default_isolation_segment(isolation_segment) }
```

Comment:
I think what Dan says is the critical part - we want to avoid enabling a scenario whereby one could entitle an org for a segment (segment.add_organization(org)) without setting that segment as the org default - this is IMHO the cleanest way to get there - you remove the ability to shoot yourself in the foot.  While I understand the philosphical/design point, I think it's throwing out the baby with the bathwater not to leverage some of what Sequel gives you when appropriate.